### PR TITLE
source-sharetribe contribution from aazam-gh

### LINKED ISSUE #46049: [connector-request] Sharetribe
This is a **suggestion** to create a connector for **Sharetribe** using the UI Connector Builder.
The website for the connector is https://www.sharetribe.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by aazam-gh**
## What

This PR adds a new connector for source Sharetribe (source-sharetribe).

Contributor-provided connector description:

Sharetribe source connector for using the sharetribe API. 
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| users | id | No pagination | ✅ |  ❌  |
| marketplace | id | No pagination | ✅ |  ❌  |
| listings | id | No pagination | ✅ |  ❌  |
| transactions | id | No pagination | ✅ |  ❌  |
| events | id | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/sharetribe.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Sharetribe
+Sharetribe source connector for using the sharetribe API.
```
Comment (marcosmarxm) [inline docs/integrations/sources/sharetribe.md:2]: Improve this description.

File: docs/integrations/sources/sharetribe.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+| `client_id` | `string` | Client ID.  |  |
+| `client_secret` | `string` | Client secret.  |  |
+| `client_refresh_token` | `string` | Refresh token.  |  |
+| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| users | id | No pagination | ✅ |  ❌  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/sharetribe.md:17]: Please confirm Pagination doesn't work to all endpoints: https://www.sharetribe.com/api-reference/#pagination

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left Comments.

**Review by natikgadzhi — DISMISSED** [review]
Transactions could in theory support incremental mode using createdAt: https://www.sharetribe.com/api-reference/marketplace.html#query-transactions

File: docs/integrations/sources/sharetribe.md (hunk 2 above)
Comment (aazam-gh) [reply docs/integrations/sources/sharetribe.md]: Added pagination support for endpoints

File: docs/integrations/sources/sharetribe.md (hunk 1 above)
Comment (aazam-gh) [reply docs/integrations/sources/sharetribe.md]: Done

**Comment by aazam-gh** [comment]
Implemented incremental sync support for streams

File: docs/integrations/sources/sharetribe.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| users | id | Default pagination | ✅ |   ✅ |
+| marketplace | id | No pagination | ✅ |  ❌  |
+| listings | id | Default pagination | ✅ |  ✅  |
+| transactions | id | Default pagination | ✅ |  ✅  |
+| events | id | Default pagination | ✅ |  ✅  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/sharetribe.md:42]: Looking into the Sharetribe website `bookings` it is an important endpoint missing in your connector.
Please include this endpoint, after let's ship it.
image

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Missing one endpoint

File: docs/integrations/sources/sharetribe.md (hunk 3 above)
Comment (marcosmarxm) [reply docs/integrations/sources/sharetribe.md:42]: The same for `reviews` and `messages`.

Comment (aazam-gh) [reply docs/integrations/sources/sharetribe.md:43]: Done!

**Comment by aazam-gh** [comment]
@marcosmarxm ready for review

**Review by marcosmarxm — APPROVED** [review]
LGTM thanks for the contribution @aazam-gh

**Pull request merged**